### PR TITLE
Feature: Generate traces following given hold/guide notes

### PR DESCRIPTION
**Pull request by GuanYixuan**
This pull request adds a right-click menu item to generate traces according to given hold/guide notes' shape and current division. This will satisfy issue #41 upon successful merge.

The menu item includes two sub-options "*add traces for*" and "*convert to traces*", which is able to create guided or non-guided trace patterns, respectively. Here's a demonstration:

- The origin and the right-click menu:
origin

- After choosing "*add traces for hold*":
Notice traces that will overlap with existing hold start/end judgements are not generated.
add

- After choosing "*convert hold to traces*":
convert

This pr only implements the feature body, while translations are not added yet.
Please feel free to make/suggest further modifications regarding the feature's name, translations, code style and so on.

### CODE REVIEW

File: MikuMikuWorld/ScoreEditorTimeline.cpp [hunk 1]
```diff
@@ -358,6 +358,14 @@ namespace MikuMikuWorld
 			                    selectedTickNum >= 3 and selectedStartNum < 2))
 				context.repeatMidsInSelection(context);
 
+			if (ImGui::BeginMenu(getString("hold_to_traces"), context.selectionHasHold())) {
```
Comment (sevenc-nanashi) [inline MikuMikuWorld/ScoreEditorTimeline.cpp:361]: Could you add `hold_to_traces` and some texts below to `.template.csv`?

File: MikuMikuWorld/ScoreContext.cpp [hunk 2]
```diff
@@ -1131,6 +1131,84 @@ namespace MikuMikuWorld
...
+
+		// beats-per-measure hardcoded to 4
+		int interval = TICKS_PER_BEAT * 4 / division;
+
+		// Here, `slide` refers to a normal hold note or a guide note
+		for (int target_slide_id : selectedNotes) {
+			if (score.notes[target_slide_id].getType() != NoteType::Hold) continue;
+
+			const HoldNote& target = score.holdNotes[target_slide_id];
+			const Note& hold_start = score.notes[target_slide_id];
```
Comment (sevenc-nanashi) [inline MikuMikuWorld/ScoreContext.cpp:1146]: could you use `camelCase` for variables? (Sorry I haven't setup linters)

**Review by sevenc-nanashi — CHANGES_REQUESTED** [review]
https://github.com/user-attachments/assets/f4b1cc24-55f5-469c-89e2-11ae6083ba50

Thank you for contribution!
But it looks like this feature creates a note which causes mmw to crash, could you fix that?

**Comment by GuanYixuan** [comment]
OK, I will take a look at these this afternoon. (It's weird that I left such an obvious bug undiscovered😂

**Comment by GuanYixuan** [comment]
It seems like the bug is caused by `score.note[]` creates empty notes when a non-exist id is passed in.
The problem should be now fixed and both code style and `.template.csv` are updated according to the request.

**Review by sevenc-nanashi — APPROVED** [review]
Sorry for late reviews; Looks good to me! Thank you for contributing.

**Pull request merged**